### PR TITLE
WSTEAMA-750: Service worker running in local development environment

### PR DESCRIPTION
**Pull request by karinathomasbbc**
Resolves JIRA https://jira.dev.bbc.co.uk/browse/WSTEAMA-750

Overall changes
======
Allows the service worker to run in local development environment

Code changes
======
- Enables service worker in local development environment
- Upgrades service worker component to component standards
- Adds unit test for the service worker

Testing
======

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

### CODE REVIEW

File: src/sw.test.js [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+  resolve(__dirname, 'service-worker-test.js'),
+  serviceWorkerCode,
+);
+
+describe('Service Worker', () => {
+  const originalFetch = global.fetch;
+  const fetchSpy = jest.spyOn(global, 'fetch');
+  let fetchEventHandler;
+
+  afterEach(() => {
```
Comment (karinathomasbbc) [inline src/sw.test.js:22]: Reminder: delete the service-worker-test.js file in `afterAll()`

File: src/app/components/ServiceWorker/index.tsx [hunk 2]
```diff
@@ -15,48 +24,35 @@ const AmpHead = () => (
...
-const ServiceWorkerContainer = () => {
+export default () => {
   const { swPath, service } = useContext(ServiceContext);
   const { isAmp, canonicalLink } = useContext(RequestContext);
-  const envIsProduction = process.env.NODE_ENV === 'production';
   const swSrc = `${process.env.SIMORGH_BASE_URL}/${service}${swPath}`;
 
   useEffect(() => {
     const shouldInstallServiceWorker =
-      process.env.SIMORGH_APP_ENV !== 'local' &&
```
Comment (karinathomasbbc) [inline src/app/components/ServiceWorker/index.tsx:34]: What was the original reason for not having this on local?

File: src/sw.test.js [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+    jest.resetAllMocks();
+    global.fetch = originalFetch;
+  });
+
+  afterAll(() => {
+    fs.unlinkSync(resolve(__dirname, 'service-worker-test.js'));
```
Comment (karinathomasbbc) [inline src/sw.test.js]: Code Climate does not appear to like this file being deleted (and causes the PR checks to fail... perhaps we should leave it - it is already git ignored

**Review by andrewscfc — APPROVED** [review]
Neat approach, ty!

**Review by emilysaffron — APPROVED** [review]
Niceeee , thank you for the knowledge share on this 🥳

**Pull request merged**